src/main/java/wolox/training/config/SecurityConfig.java

```diff
@@ -0,0 +1,41 @@
+package wolox.training.config;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.config.annotation.authentication.builders.AuthenticationManagerBuilder;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.EnableWebSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import wolox.training.authentication.UserAndPasswordAuthenticationProvider;
+
+@Configuration
+@EnableWebSecurity
+public class SecurityConfig extends WebSecurityConfigurerAdapter {
+
+    @Autowired
+    private UserAndPasswordAuthenticationProvider userAndPasswordAuthenticationProvider;
+
+    @Override
+    protected void configure(HttpSecurity http) throws Exception {
+        http
+            .csrf()
+            .disable()
+            .exceptionHandling()
+            .and()
+            .authorizeRequests()
+            .antMatchers(HttpMethod.POST, "/api/books/").permitAll()
```

Comment:
You can apply authorization request in other way. There is a common pattern in all protected urls. Look at this [example](http://blog.florian-hopf.de/2017/08/spring-security.html)

Reply:
Do you mean overriding the `configure(WebSecurity web) throws Exception`?